src/Essential.Diagnostics.RollingFileTraceListener/IO/FileSystem.cs

```diff
@@ -7,6 +7,13 @@ namespace Essential.IO
     /// </summary>
     public class FileSystem : IFileSystem
     {
+        /// <summary>
+        ///     Gets or sets the value indicating whether all subdirectories in full file path
+        ///     should be checked for existence and re-created if missed
+        ///     before opening the file. Default value is <c>False</c>.
+        /// </summary>
+        public bool CreateSubdirectories { get; set; }
```

Comment:
As noted in the description of FileSystem / IFileSystem, it is meant to be a wrapper around System.IO.File to allow substitution during testing.

It should not be extended with new properties that implement business logic, such as CreateSubdirectories.  The Create property should be on the RollingTextWriter class, and it control the logic of which File methods to call.